### PR TITLE
[plugin.video.plutotv] 1.0.5

### PR DESCRIPTION
**Pull request by Lunatixz**
https://github.com/xbmc/repo-scripts/pull/535

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by enen92** [comment]
@Lunatixz fyi with the latest version of the plugin I get a script error (the addon does not fail gracefully) if no credentials are set:

```
14:39:06.751 T:123145470595072   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.TypeError'>
                                            Error Contents: 'NoneType' object is not iterable
                                            Traceback (most recent call last):
                                              File "/Users/enen/Library/Application Support/Kodi/addons/plugin.video.plutotv/default.py", line 449, in <module>
                                                if mode==None:  PlutoTV().mainMenu()
                                              File "/Users/enen/Library/Application Support/Kodi/addons/plugin.video.plutotv/default.py", line 100, in __init__
                                                self.categoryMenu = self.getCategories()
                                              File "/Users/enen/Library/Application Support/Kodi/addons/plugin.video.plutotv/default.py", line 188, in getCategories
                                                for channel in data:
                                            TypeError: 'NoneType' object is not iterable
                                            -->End of Python script error report<--
```

File: plugin.video.plutotv/resources/settings.xml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <?xml version="1.0" encoding="utf-8" standalone="yes"?>
 <settings>
-    <setting id="User_Email"       type="text" label="30001" default="" />
-    <setting id="User_Password"    type="text" label="30002" default="" />
-    <setting id="Hide_Ads"         type="bool" label="30003" default="false" />
-    <setting id="Filter_Region"    type="bool" label="30004" default="false" />
-    <setting id="Enable_Debugging" type="bool" label="30000" default="false" />
+    <setting id="User_Email"       type="text"   label="30001" default="" />
+    <setting id="User_Password"    type="text"   label="30002" default="" />
+    <setting id="Hide_Ads"         type="bool"   label="30003" default="false" />
```
Comment (enen92) [inline plugin.video.plutotv/resources/settings.xml]: @Lunatixz unfortunately we talked internally and agreed addons on the repo should not try to bypass ads if there is a way for them to be played (unless the original app does it too). Can you please drop this option from the addon?

**Comment by Lunatixz** [comment]
Yeah I took care of it... Feel free to test out 1.0.5 with default settings, I added user visual aids.

File: plugin.video.plutotv/resources/settings.xml (hunk 1 above)
Comment (Lunatixz) [inline plugin.video.plutotv/resources/settings.xml]: Sure, I'll remove it.

Comment (Lunatixz) [inline plugin.video.plutotv/resources/settings.xml]: Is it okay if I leave it an internal command only accessible when running along with a script, in this case PseudoTV Live. BTW I plan on submitting PseudoTV Live in the future.

Comment (enen92) [inline plugin.video.plutotv/resources/settings.xml]: What is the problem about having ads when using PseudoTV Live?

**Comment by Lunatixz** [comment]
updated module submission here: https://github.com/xbmc/repo-scripts/pull/537

**Comment by Lunatixz** [comment]
i'll have to rework the code the work without the module. Thx for the review :)

**Pull request closed without merging**